apps/gimbtonomy/LEDMatrix.cpp

```diff
@@ -0,0 +1,47 @@
+/* LED Matrix Source File */
+
```

Comment:
@jetkov  this lib should be non-blocking. my thought is that the api functions should simply begin an asynchronous process and return a bool relating to whether the async process began. I think the user of this lib should also pass in a callback function which the async process calls when it is completely done. Specifically, the callback would be sending a ack CAN msg. This should be handeled at the level of main.cpp, hence the callback. do you think this a good idea.

Reply:
For just setting the PWM duty cycles we don't need it to be async as it is essentially already async (it is very quick). For the flashing which I dont think is currently done, that can spin up a thread. @laineee pwm frequency should be used for brightness, but theres the actual flashing animation frequency which is different, that will need to be done with a thread/timer loop. I can explain more in the next meeting.